sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -191,50 +190,544 @@ def get_next(next_link=None):
+
+            pipeline_response = self._client._client._pipeline.run(  # pylint: disable=protected-access
+                request, stream=False, **kwargs
+            )
+            response = pipeline_response.http_response
+
+            if response.status_code not in [200]:
+                error = self._client._deserialize.failsafe_deserialize(  # pylint: disable=protected-access
+                    AcrErrors, response
+                )
+                map_error(status_code=response.status_code, response=response, error_map=error_map)
+                raise HttpResponseError(response=response, model=error)
+
+            return pipeline_response
+
+        return ItemPaged(get_next, extract_data)
+
+    @distributed_trace
+    def delete_manifest(self, repository, tag_or_digest, **kwargs):
+        # type: (str, str, **Any) -> None
+        """Delete a manifest
+
+        :param str repository: Name of the repository the manifest belongs to
+        :param str tag_or_digest: Tag or digest of the manifest to be deleted
+        :returns: None
+        :rtype: None
+        :raises: :class:`~azure.core.exceptions.ResourceNotFoundError`
+
+        Example
+
+        .. code-block:: python
+            from azure.containerregistry import ContainerRepositoryClient
+            from azure.identity import DefaultAzureCredential
+            account_url = os.environ["CONTAINERREGISTRY_ENDPOINT"]
+            client = ContainerRepositoryClient(account_url, "my_repository", DefaultAzureCredential())
+            client.delete()
+        """
+        if _is_tag(tag_or_digest):
+            tag_or_digest = self._get_digest_from_tag(repository, tag_or_digest)
+
+        try:
+            self._client.container_registry.delete_manifest(repository, tag_or_digest, **kwargs)
+        except ResourceNotFoundError:
+            pass
+
+    @distributed_trace
+    def delete_tag(self, repository, tag, **kwargs):
+        # type: (str, str, **Any) -> None
+        """Delete a tag from a repository
+
+        :param str repository: Name of the repository the tag belongs to
+        :param str tag: The tag to be deleted
+        :returns: None
+        :rtype: None
+        :raises: :class:`~azure.core.exceptions.ResourceNotFoundError`
+
+        Example
+
+        .. code-block:: python
+            from azure.containerregistry import ContainerRepositoryClient
+            from azure.identity import DefaultAzureCredential
+            account_url = os.environ["CONTAINERREGISTRY_ENDPOINT"]
+            client = ContainerRepositoryClient(account_url, "my_repository", DefaultAzureCredential())
+            for artifact in client.list_tags():
+                client.delete_tag(tag.name)
+        """
+        try:
+            self._client.container_registry.delete_tag(repository, tag, **kwargs)
+        except ResourceNotFoundError:
+            pass
+
+    @distributed_trace
+    def get_manifest_properties(self, repository, tag_or_digest, **kwargs):
```

Comment:
In the review we suggested that in general tag should default to `latest` - but I don't see that used anywhere?

Reply:
I am looking into whether this is recommended by the docker community before adding it. There's a bit of a mixed feeling in the community about whether this is a good idea. One example was a situation where a customer is deploying from a registry your "latest" tag, but while this deployment is made (deployments can take a long time, similar to a live pipeline run) there is an inadvertent change that doesn't specify the tag, which defaults to "latest" and can interfere with a deployment so now the customer is not deploying the repository they intended.